### PR TITLE
Add support for TS011F (_TZ3000_cphmq0q7) smart plug

### PR DESCRIPTION
**Pull request by yratanov**
Inspired by TS0121_plug, with slight change.

https://github.com/Koenkk/zigbee2mqtt/issues/7463
https://github.com/Koenkk/zigbee2mqtt/issues/7405
https://github.com/Koenkk/zigbee-herdsman-converters/issues/2557

image
image

### CODE REVIEW

**Comment by oetelaar24** [comment]
Hi, @yratanov ,

Thanks for this, you also mentioned my thread. As soon as I saw your post, I merged your change to my local tuya.js file.

However I seem to be getting an error when it tries to configure the plug. Otherwise, it does seem to work well! Which is great!

The error is:
```
Zigbee2MQTT:error 2021-05-14 06:02:13: Failed to configure '0x847127fffe8d9579', attempt 3 (TypeError: Cannot read property 'configureKey' of undefined
,    at Configure.configure (/app/lib/extension/configure.js:138:69))
```

**Comment by yratanov** [comment]
@oetelaar24 yep, I've seen this error, and I added 
```js
    meta: {
        configureKey: 1,
    },
```
locally, but it seems it's an outdated method, so I'm not sure how to fix it properly.

**Comment by oetelaar24** [comment]
Ah yes, I see this method being used by others as well. It does resolve the error. 
I see it mentioned here a while back: https://github.com/Koenkk/zigbee2mqtt/issues/1758

File: devices/tuya.js [hunk 1]
```diff
@@ -626,6 +626,45 @@ module.exports = [
...
+            const endpoint = device.getEndpoint(1);
+            await reporting.bind(endpoint, coordinatorEndpoint, ['genOnOff', 'haElectricalMeasurement']);
+            endpoint.saveClusterAttributeKeyValue('seMetering', {divisor: 100, multiplier: 1});
+            endpoint.saveClusterAttributeKeyValue('haElectricalMeasurement', {
+                acVoltageMultiplier: 1, acVoltageDivisor: 1, acCurrentMultiplier: 1, acCurrentDivisor: 1000, acPowerMultiplier: 1,
+                acPowerDivisor: 1,
+            });
+        },
+        // This device doesn't support reporting correctly.
+        // https://github.com/Koenkk/zigbee-herdsman-converters/pull/1270
```
Comment (Koenkk) [inline devices/tuya.js:646]: Are you sure this device does not support reporting? Can you check if power reporting also works when:
- Removing the `onEvent`
- Replacing configure with:

```js
configure: async (device, coordinatorEndpoint, logger) => {
    const endpoint = device.getEndpoint(1);
    await reporting.bind(endpoint, coordinatorEndpoint, ['genOnOff', 'haElectricalMeasurement']);
    await reporting.rmsVoltage(endpoint);
    await reporting.rmsCurrent(endpoint);
    await reporting.activePower(endpoint);
    await reporting.readEletricalMeasurementMultiplierDivisors(endpoint);
},
```

File: devices/tuya.js [hunk 2]
```diff
@@ -626,6 +626,45 @@ module.exports = [
             }
         },
     },
+    {
+        fingerprint: [{modelID: 'TS011F', manufacturerName: '_TZ3000_cphmq0q7'},],
+        model: 'TS011F',
+        description: 'Smart EU Plug',
+        vendor: 'TuYa',
+        fromZigbee: [fz.on_off, fz.electrical_measurement, fz.metering, fz.ignore_basic_report, fz.tuya_switch_power_outage_memory],
```
Comment (Koenkk) [inline devices/tuya.js:634]: Do you get an updated `state` when controlling the plug via the button on it?

**Comment by Koenkk** [comment]
> @oetelaar24 yep, I've seen this error, and I added
> 
> ```js
>     meta: {
>         configureKey: 1,
>     },
> ```
> 
> locally, but it seems it's an outdated method, so I'm not sure how to fix it properly.

Configure key isn't needed anymore since Z2M 1.18.3**-dev**

File: devices/tuya.js (hunk 2 above)
Comment (yratanov) [reply devices/tuya.js:634]: Yes

File: devices/tuya.js (hunk 1 above)
Comment (yratanov) [reply devices/tuya.js]: Thank you! It does work like that. I've pushed the changes

**Comment by yratanov** [comment]
@Koenkk 
> Configure key isn't needed anymore since Z2M 1.18.3-dev (it is in 1.18.3)

It's weird, because I have 1.18.3-1.  But I'm adding this code as external converter, as described [here](https://www.zigbee2mqtt.io/how_tos/how_to_support_new_devices.html), may be it's somehow different.

**Comment by Koenkk** [comment]
@yratanov `1.18.3-1` != `1.18.3-dev`, for dev use the edge version of the addon.

File: devices/tuya.js (hunk 2 above)
Comment (Koenkk) [reply devices/tuya.js:634]: perfect

File: devices/tuya.js (hunk 1 above)
Comment (yratanov) [reply devices/tuya.js]: I was wrong. It doesn't update. I'll revert and create another PR

**Comment by McGiverGim** [comment]
Only as information, I've this device, seems to work, but the image is totally different:
https://www.aliexpress.com/item/1005002495230651.html
image

I suppose it not matters...

**Comment by Koenkk** [comment]
@McGiverGim I received exactly this one indeed. Picture is currently not correct but will be fixed with the next release.

**Comment by oetelaar24** [comment]
It soes seem to work well. However, and I'm unsure why it's happening, some of the plugs have started to turn off at a seemingly random moment.. Which is kindoff a bummer when the device behind it needs to be on...
It could be the plug is faulty ofcourse, but I'm not sure how I can see that...

**Comment by jusitnhu14** [comment]
Hi was wondering if someone can point me to the external converter MQTT for Zigbee Model
TS011F
Zigbee Manufacturer
_TZ3000_myruaaxg
Its a TUYA 1 gang plug and USB charging port

Thanks

**Pull request merged**